src/Fantomas.Core.Tests/ChainTests.fs

```diff
@@ -299,3 +299,52 @@ Map
 Map.empty<_, obj>
     .Add("headerAction", modifyHeader.Action.ArmValue)
 """
+
+[<Test>]
+let ``all simple links should be on the same line, 2712`` () =
+    formatSourceString
+        false
+        """
+type Duck() =
+    member this.Duck  = Duck ()
+    member this.Goose() = Duck()
+    
+let d = Duck()
+
+d.Duck.Duck.Duck.Goose().Duck.Goose().Duck.Duck.Goose().Duck.Duck.Duck.Goose().Duck.Duck.Duck.Duck.Goose()
+"""
+        config
+    |> prepend newline
+    |> should
+        equal
+        """
+type Duck() =
+    member this.Duck = Duck()
+    member this.Goose() = Duck()
+
+let d = Duck()
+
+d.Duck.Duck.Duck
+    .Goose()
+    .Duck.Goose()
+    .Duck.Duck.Goose()
+    .Duck.Duck.Duck.Goose()
+    .Duck.Duck.Duck.Duck.Goose()
+"""
+
+[<Test>]
+let ``very long chain with a some index expressions`` () =
```

Comment:
@dawedawe this one is a bit tricky. The whole chain does not fit within the `MaxDotGetExpressionWidth` boundary, but there is no application to chop it logically.

Reply:
It follows the code path where the leading links are all the links in
```fsharp
match links with
| [] ->
    fun ctx ->
        isShortExpression
            ctx.Config.MaxDotGetExpressionWidth
            short
            (match leadingChain with
             | [] -> sepNone
             | head :: links -> genFirstLinkAndIndentOther head links)
            ctx
```

The `head :: links -> genFirstLinkAndIndentOther head links` will print the first on a single line as a last resort.

This feels a bit distorted. Thoughts?